src/Server.cpp

```diff
@@ -449,7 +449,16 @@ void cServer::ExecuteConsoleCommand(const AString & a_Cmd, cCommandOutputCallbac
 	}
 	else if (split[0] == "reload")
 	{
-		cPluginManager::Get()->ReloadPlugins();
+		if (split.size() > 1)
+		{
+			cPluginManager::Get()->UnloadPlugin(split[1]);
+			a_Output.Out("Plugin unloaded scheduled");
+			a_Output.Out(cPluginManager::Get()->LoadPlugin(split[1]) ? "Plugin loaded" : "Error occurred loading plugin");
```

Comment:
Because unloading a plugin requires locking a mutex and since the console handler is already holding another mutex, it resulted in deadlocks in certain scenarios.